### PR TITLE
Feature/key value store

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Add key/value store scope by current API user (and current tenant at the same time)

# Impacts
API frontend

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [X] Document the feature
- [X] Add unit tests
- [n/a] Add acceptance tests (TestLink)

### CODE REVIEW

**Comment by julsemaan** [comment]
@satkunas 

Can we prioritize the review of this, I'll be needing some AAA features for another branch sooner than later so having this in devel will help a lot

File: docs/api/spec/paths/preference.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        description: Get item
+      '400':
+        $ref: '#/components/responses/BadRequest'
+      '404':
+        $ref: '#/components/responses/NotFound'
+      '422':
+        $ref: '#/components/responses/UnprocessableEntity'
+  put:
+    description: Replace an item
+    operationId: api.v1.Ip4logs.replace
```
Comment (satkunas) [inline docs/api/spec/paths/preference.yaml]: Invalid operationId

**Comment by satkunas** [comment]
CLI use appears to not have a default user. This would help facilitate testing.

`/usr/local/pf/sbin/pfperl-api get -M GET /api/v1/preferences | jq ''`
```json
{
  "errors": [],
  "message": "Username not present in request",
  "status": 404
}
```

**Comment by julsemaan** [comment]
Yes there is no default user, that is intended, you need to go through the API frontend for this to work or set "X-PacketFence-Username" manually in your command

**Review by satkunas — CHANGES_REQUESTED** [review]

**Comment by satkunas** [comment]
I would like to see the GET response changed from:

``json
{
  "items": [
    {
      "id": "testkey",
      "value": "testvalue2"
    }
  ],
  "status": 200
}
```

to 

```json
{
  "items": [
    {
      "id": "testkey",
      "value": "testvalue2"
    }
  ],
  "status": 200
}
```

**Pull request merged**